src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementService.java

```diff
@@ -259,4 +263,10 @@ private boolean isBeforeOrEqual(LocalDate date, LocalDate compareToDate) {
         return !compareToDate.isAfter(date);
     }
 
+    private boolean hasExistingConfirmationSubmission (Transaction transaction) {
+        if (transaction.getResources() != null) {
+            return transaction.getResources().entrySet().stream().anyMatch(resourceEntry -> resourceEntry.getValue().getKind().equals("confirmation-statement"));
```

Comment:
resourceEntry -> resourceEntry.getValue().getKind().equals("confirmation-statement")  could throw a null pointer if kind is null, can you try reversing the string check like this
resourceEntry -> "confirmation-statement".equals(resourceEntry.getValue().getKind())